spike-ticket/src/main/java/com/in2it/spiketicket/repository/TicketRepository.java

```diff
@@ -1,34 +1,42 @@
 package com.in2it.spiketicket.repository;
 
-import java.time.LocalDate;
+import java.util.Optional;
 
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
-import com.in2it.spiketicket.entity.Ticket;
 import com.in2it.spiketicket.constants.Status;
-import java.util.List;
-import java.util.Optional;
-
+import com.in2it.spiketicket.entity.Ticket;
 
+@Repository
+public interface TicketRepository extends JpaRepository<Ticket, Long>, JpaSpecificationExecutor<Ticket> {
 
+	Page<Ticket> findByAssignedBy(String assignBy, Pageable pageable);
 
-@Repository
-public interface TicketRepository extends JpaRepository<Ticket, Long> {
-	
-	Page<Ticket> findByIdEqualsOrTitleContainingIgnoreCaseOrAssignToContainingIgnoreCaseOrAssignedByContainingIgnoreCaseOrCreatedAtOrStatusContainingAllIgnoreCaseAndDeleted(
-            Long id, String title, String assignTo, String assignedBy, LocalDate createdAt,String status,boolean deleted, Pageable pageable);
-	Page<Ticket>  findByAssignedBy(String assignBy, Pageable pageable);
 	Page<Ticket> findByDeleted(boolean deleted, Pageable pageable);
-	
```

Comment:
what i asked was to check the implementation of criteria query as it can provide flexibility of dynamic columns search , and  here using derived or native query make same effort with static columns.